crates/tap-agent/src/tap/context/receipt.rs

```diff
@@ -187,6 +197,41 @@ impl ReceiptDelete for TapAgentContext {
     }
 }
 
+/// Implements a [ReceiptRead] for [TapReceipt]
+/// in case [super::NetworkVersion] is [Horizon]
+///
+/// This is important because receipts for each network version
+/// are stored in a different database table
+#[async_trait::async_trait]
+impl ReceiptRead<TapReceipt> for TapAgentContext<Horizon> {
+    type AdapterError = AdapterError;
+
+    async fn retrieve_receipts_in_timestamp_range<R: RangeBounds<u64> + Send>(
+        &self,
+        _timestamp_range_ns: R,
+        _receipts_limit: Option<u64>,
+    ) -> Result<Vec<CheckingReceipt>, Self::AdapterError> {
+        unimplemented!()
+    }
+}
+
+/// Implements a [ReceiptDelete] for [TapReceipt]
+/// in case [super::NetworkVersion] is [Horizon]
+///
+/// This is important because receipts for each network version
+/// are stored in a different database table
+#[async_trait::async_trait]
```

Comment:
We need async-trait here for object-safety, is that right?

Reply:
We need it because `ReceiptRead` needs. The lib was created at a time when async methods weren't supported by traits in Rust.

Reply:
Great context, thanks!

Reply:
Do you want me to add this in the documentation?